packages/react-avatar/SPEC.md

```diff
@@ -0,0 +1,243 @@
+
+#### Display
+
+Both the v8 and v0 components support displaying an image, initials, and an icon.
+
+The v8 `Persona` also supports the full name and extra detail text to the right of the image. But v8 appears to not support a custom icon, and can only show the default person icon, or a "?" icon
+
+#### Status/Presence Badge
+
+Both components support showing a badge to indicate status, but they have different APIs. The v8 `Persona` has an enum of preset options for the presence badge. The v0 `Avatar` supports rendering a `Status` component, which is more customizable and can render an icon, and specify different sizes and colors.
+
+## Sample Code
+
+Basic examples:
+
+```jsx
+<Avatar name="Miguel Garcia" />
+<Avatar size={72} name="Mona Kane" image="./MonaKane.jpg" />
+<Avatar square icon={<IDBadgeIcon />} />
+```
+
+Displaying a badge:
+
+```jsx
+<Avatar name="Allan Munger" badge={<PresenceBadge status="busy">} />
+```
+
+With active state indication:
+
+```jsx
+<Avatar name="Daisy Phillips" active={true} activeDisplay="ring-shadow" />
+<Avatar name="Robin Counts" active={false} activeDisplay="ring-shadow" />
+```
+
+## Variants
+
+N/A
+
+## API
+
+From [Avatar.types.tsx](https://github.com/microsoft/fluentui/blob/master/packages/react-avatar/src/components/Avatar/Avatar.types.tsx):
+
+### Slots
+
+- `image` - The Avatar's image, if available.
+- `label` - The text shown when there's no image. Defaults to the initials derived from `name` using `getInitials`.
+- `icon` - Icon displayed when there's no image or intials available.
+- `badge` - Badge to show the avatar's status.
+
+### Props
+
+```ts
+export interface AvatarProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+  /** The Avatar's image. */
+  image?: ShorthandValue<{}>;
+
+  /** The label shown when there's no image. Defaults to the initials derived from `name` using `getInitials`. */
+  label?: ShorthandValue<{}>;
+
+  /** Icon displayed when there's no image or intials available, or if `display="icon"`. */
+  icon?: ShorthandValue<{}>;
+
+  /** What the avatar displays. This can be used to override the default behavior, for example to show the icon even
+   *  if the avatar has initials that could be shown as the label. */
+  display?: 'image' | 'label' | 'icon';
+
+  /**
+   * Optional activity indicator
+   * * true: the avatar will be decorated according to activeDisplay
+   * * false: the avatar will be reduced in size and partially transparent
+   * * undefined (default): normal display
+   */
+  active?: boolean;
+
+  /**
+   * The type of visual treatment to use when `active="true"`
+   * @defaultvalue ring
+   */
+  activeDisplay?: 'ring' | 'shadow' | 'glow' | 'ring-shadow' | 'ring-glow';
+
+  /** Badge to show the avatar's status. */
+  badge?: ShorthandValue<BadgeProps>;
+
+  /** The name used for displaying the initials of the avatar if the image is not provided. */
+  name?: string;
+
+  /** The avatar can have a square shape. */
+  square?: boolean;
```

Comment:
there is a corner case where hexagonal avatar needs to be displayed, in v0 we support it, but not directly in the Avatar - as a global svg object needs to be present in the DOM - https://github.com/microsoft/fluentui/blob/master/packages/fluentui/react-northstar-prototypes/src/prototypes/hexagonalAvatar/index.tsx

We will just need to make sure this will be still supported in converged Avatar

Reply:
Yes, this is supported by using a custom background SVG, by setting `tokens.background` to an SVG. I'll call it out in the spec.

You can see a demo of it here:
http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/heads/master/react-avatar/storybook/index.html?path=/story/avatar--custom-shape

E.g.:
```jsx
        <Avatar
          icon={<ChatBotIcon />}
          display="icon"
          tokens={{
            width: 'calc(var(--avatar-height) * 1.125)',
            background: `url('${examples.hexagon}') 0px/contain no-repeat`,
            borderRadius: '0',
          }}
        />
```